test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonWriterAsyncBaseTests.cs

```diff
@@ -0,0 +1,243 @@
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeDouble""");
+                await jsonWriter.WriteValueAsync(10.2);
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeFloat""");
+                await jsonWriter.WriteValueAsync(10.2f);
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeDecimal""");
+                await jsonWriter.WriteValueAsync(10.2m);
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeGuid""");
+                await jsonWriter.WriteValueAsync(new Guid("00000012-0000-0000-0000-012345678900"));
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeObject""");
+                await jsonWriter.StartObjectScopeAsync();
+                await jsonWriter.EndObjectScopeAsync();
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeDateTimeOffset""");
+                await jsonWriter.WriteValueAsync(new DateTimeOffset(2014, 12, 31, 12, 42, 30, new TimeSpan(1, 20, 0)));
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeDate""");
+                await jsonWriter.WriteValueAsync(new Date(2014, 12, 31));
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeTimeOfDay""");
+                await jsonWriter.WriteValueAsync(new TimeOfDay(12, 42, 30, 100));
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.StartArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeTimeSpan""");
+                await jsonWriter.WriteValueAsync(new TimeSpan(12, 42, 30));
+                await jsonWriter.EndArrayScopeAsync();
+                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeObject""");
+                await jsonWriter.StartObjectScopeAsync();
+                await jsonWriter.EndObjectScopeAsync();
+                await jsonWriter.EndArrayScopeAsync();

```

Comment:
Did you really intend this:
```
[[\"rawAtArrayStartBeforeTimeSpan\",\"PT12H42M30S\"],\"rawAtArrayStartBeforeObject\",{}]
```
or this:
```
[\"rawAtArrayStartBeforeTimeSpan\",\"PT12H42M30S\"],[\"rawAtArrayStartBeforeObject\",{}]
```
If you intended the latter, _and my guts tells me that's what you intended_, I'd expect the code here to look like:
```suggestion
                await jsonWriter.StartArrayScopeAsync();
                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeTimeSpan""");
                await jsonWriter.WriteValueAsync(new TimeSpan(12, 42, 30));
                await jsonWriter.EndArrayScopeAsync();
                await jsonWriter.StartArrayScopeAsync();
                await jsonWriter.WriteRawValueAsync(@"""rawAtArrayStartBeforeObject""");
                await jsonWriter.StartObjectScopeAsync();
                await jsonWriter.EndObjectScopeAsync();
                await jsonWriter.EndArrayScopeAsync();
```

Reply:
I think I did intentend the latter, i.e. to have the raw value at the start of the array before the object.